core/modules/link/link.module

```diff
@@ -930,6 +945,67 @@ function link_field_process($element, $form_state, $complete_form) {
   return $element;
 }
 
+/**
+ * Link field element validation.
+ */
+function link_field_element_validate(&$element, &$form_state, $complete_form) {
+  $value = $element['#value'];
+  $is_default_field_value = FALSE;
+
+  // Check if this element is used in Field API Link field.
+  if (isset($element['#link_field_options']['field_name'])) {
```

Comment:
I don't see where `$element['#link_field_options']['field_name']` is ever set (or documented). Does this situation actually occur?